toolkit/scripts/containerized-build/resources/setup_functions.sh

```diff
@@ -0,0 +1,146 @@
+#! /bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+TOPDIR=<TOPDIR>
+SPECS_DIR=$TOPDIR/SPECS
+SOURCES_DIR=$TOPDIR/SOURCES
+RPMS_DIR=$TOPDIR/RPMS
+SRPMS_DIR=$TOPDIR/SRPMS
+IS_REPO_ENABLED=false
+
+# Mariner macro files used during spec parsing (as defined in toolkit/scripts/rpmops.sh)
+DEFINES=(-D "with_check 1")
+MACROS=()
+for macro_file in "$SPECS_DIR"/mariner-rpm-macros/macros* "$SPECS_DIR"/pyproject-rpm-macros/macros.pyproject "$SPECS_DIR"/perl/macros.perl
+do
+  MACROS+=("--load=$macro_file")
+done
+
+# Create symlink from SPECS/ to SOURCES/ when rpm is called
+rpm() {
+    local args=("$@")
+    command "$FUNCNAME" "${args[@]}"
+    if [[ ${args} = *"i"* ]]; then
+        for ((i = 0; i < ${#args[@]}; ++i)); do
+            if [[ ${args[$i]} = *".src.rpm"* ]]; then
+                local SPEC=${args[$i]}
+                SPEC=${SPEC##*/}
+                SPEC=${SPEC%-*}
+                SPEC=${SPEC%-*}
+                SPEC=${SPEC%\**}
+                mkdir -p $SOURCES_DIR
+                ln -sf $SPECS_DIR/$SPEC/* $SOURCES_DIR/
+            fi
+        done
+    fi
+    rm -f $SPECS_DIR/*.spec
+    rm -f $SOURCES_DIR/*.spec
+    rm -f $SOURCES_DIR/*.signatures.json
+}
+
+# Installs srpm, pkg dependencies and builds pkg
+build_pkg() {
+    local PKG=("$@")
+    if [ -z "$PKG" ]; then echo "Please provide pkg name"; return; fi
+    rpm -ihv /mnt/INTERMEDIATE_SRPMS/$PKG*.src.rpm
+    $(install_dependencies $PKG)
+    rpmbuild -ba $SPECS_DIR/$PKG/$PKG.spec
+}
+
+# Show help on useful commands
+show_help() {
+    cat /mariner_setup_dir/welcome.txt
+    cat /mariner_setup_dir/mounts.txt
+    echo -e "* \n* Local repo information:"
+    if [[ "${IS_REPO_ENABLED}" == "true" ]]; then
+        echo -e "*\tLocal repo is enabled. Package dependencies will be installed from $RPMS_DIR, /repo and upstream server"
+    else
+        echo -e "*\tLocal repo is not enabled. Package dependencies will be installed from upstream server"
+    fi
+    echo "******************************************************************************************"
+}
+
+# Refresh repo cache with newly built RPM, use Mariner specific DEFINES
+rpmbuild() {
```

Comment:
Just for my understanding, what is the purpose of this rpmbuild function? If it is to simply refresh the local repo cache, then it doesn't seem like it's doing any rpmbuild operations.

Reply:
This is adding functionality to the already defined function rpmbuild, by supplying Mariner 'DEFINES' and then refreshing the repo cache. It is executing the function at line 68: command "$FUNCNAME" "${DEFINES[@]}" "${args[@]}"

Reply:
Ah I had missed `command "$FUNCNAME" "${DEFINES[@]}" "${args[@]}"`. This is our wrapper around `rpmbuild` to do a little more processing. Thanks for the explanation!